src/lib/falcor/falcorUtils.js

```diff
@@ -28,3 +28,73 @@ export function setAppReady() {
   console.log("APP SET AS READY")
   appReady = true
 }
+
+// Below function based on function from: https://github.com/ekosz/falcor-expand-cache
+// MIT License
+// Copyright (c) 2015 Eric Koslow
+export function expandCache(cache) {
+  function followPath(path) {
+    // If using dot notation obj.key.key.key
+    if (typeof path === "string") {
+      path = path.split('.');
+    }
+    return path.reduce((acc, part) => {
+      if (acc !== undefined && acc.hasOwnProperty(part)) {
+        return acc[part];
+      }
+
+      return undefined;
+    }, cache);
+  }
+
+  function isObject(val) {
+    if (val === null || val instanceof Array) return false;
+    return typeof val === "object";
+  }
+
+  // If empty return itself
+  if (!cache) return cache;
+  // Expanding
+  let stack = [];
+  Object.keys(cache).forEach((key) => {
+    stack.push([key]);
+  })
+  while (stack.length > 0) {
+    let pathArray = stack.pop();
+    // Parent also works for array length 1, aka initial keys
+    // Used for assigning
+    let parent = followPath(pathArray.slice(0, pathArray.length-1));
+    let key = pathArray[pathArray.length-1];
+    // Since key was pushed on to stack from objects keys, it will exist
+    let val = parent[key];
+    if (val === undefined) {
+      throw new Error("Missing part of JSON graph in expand");
+    }
+    if (!isObject(val)) {
+      continue;
+    }
+    else if (val.$type) {
+      switch (val.$type) {
+        case "atom":
+          parent[key] = val.value;
+          break;
+        case "error":
+          parent[key] = new Error(val.value);
+          break;
+        case "ref":
+          parent[key] = followPath(val.value);
+          break;
+        default:
+          return undefined;
+      }
+    }
+    else{
+      Object.keys(val).forEach((key) => {
+        let next = pathArray.slice();
+        next.push(key);
```

Comment:
When I search the net it looks like people agree on this being the standard way though?